tools/documentation_developer/how_to_contribute.rst

```diff
@@ -1,6 +1,607 @@
+
+Coding Style
+^^^^^^^^^^^^
+
+The HiDimStat codebase follows `PEP8 <https://peps.python.org/pep-0008/>`__ styling.
+
+
+The main conventions we enforce are :
+
+- line length < 80
+- spaces around operators
+- meaningful variable names
+- function names are underscore separated (e.g., ``a_nice_function``) and as short as possible
+- public functions exposed in their parent module's init file
+- private function names preceded with a "_" and very explicit
+- classes in CamelCase
+- 2 empty lines between functions or classes
+- 4 spaces instead of tabs per indentation level
+- 
+
+You can check that any code you may have edited follows these conventions
+by running `black <https://black.readthedocs.io/en/stable/>`__.
+
+
+Testing the code
+^^^^^^^^^^^^^^^^
+
+Tests for files in a given folder should be located in the folder named ``tests`` 
+with a tree structure that reflects the source folder. For example, the tests 
+for the functions in `src/hidimstat/statistical_tools/p_values.py` are in 
+`test/statistical_tools/test_p_values.py`. 
+
+Tests should check all functionalities of the code that you are going to
+add. If needed, additional tests should be added to verify that other
+objects behave correctly.
+
+.. If there is any dependance between the tests or function, we need to write
+   an example of it here. 
+
+Run each updated test file using ``pytest``
+(`pytest docs <https://docs.pytest.org/en/stable>`_):
+
+.. code:: sh
+
+   pytest -vsl tests/test_amazing_method.py
+
+The ``-vsl`` flag provides more information when running the tests.
+
+It is also possible to run a specific test, or set of tests using the
+commands ``pytest the_file.py::the_test``, or
+``pytest the_file.py -k 'test_name_pattern'``. This is helpful to avoid
+having to run all the tests.
+
+If you work on Windows, you might have some issues with the working
+directory if you use ``pytest``, while ``python -m pytest ...`` should
+be more robust.
+
+Once you are satisfied with your changes, you can run all the tests to make sure
+that your change did not break code elsewhere:
+
+.. code:: sh
+
+    pytest -s tests
+
+Finally, sync your changes with the remote repository and wait for CI to run.
+
+Checking coverage on the local machine
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Checking coverage is one of the operations that is performed after
+submitting the code. As this operation may take a long time online, it
+is possible to check whether the code coverage is high enough on your
+local machine.
+
+Run your tests with the ``--cov`` and ``--cov-report`` arguments:
+
+.. code:: sh
+
+   pytest -vsl tests/test_amazing_method.py --cov=. --cov-report=html
+
+This will create the folder ``htmlcov``: by opening
+``htmlcov/index.html`` it is possible to check what lines are covered in
+each file.
+
+.. ADD THIS SECTION WHEN THERE WILL BE DOCSTRING TESTS
+``` (I don't think there will be any in a near future)
```

Comment:
I want to include some tests related to the figure generated in the examples.
For it's important because the change in the figures can imply a change in the text of the examples. 
I open an issue regard to it #312 .